pkg/manager/member/tidbcluster_status_manager.go

```diff
@@ -125,6 +136,84 @@ func (m *TidbClusterStatusManager) syncDashboardMetricStorage(tc *v1alpha1.TidbC
 	return nil
 }
 
+// ref https://github.com/pingcap/tidb/blob/36b04d1aa01db722b3f07af759168c6b8da33801/domain/infosync/info.go#L72
+// search `TopologyInformationPath` about how the key with 'ttl' and 'info' suffix is updated in that file.
+func getStaleTidbInfoKey(ctx context.Context, client pdapi.PDEtcdClient) (staleKeys []*pdapi.KeyValue, err error) {
+	kvs, err := client.Get(tidbPrefix, true /*prefix*/)
```

Comment:
The key we get here should be for this cluster only:

- /topology/tidb/basic-tidb-0.basic-tidb-peer.tidb-cluster.svc:4000/info (to match the TiDB sts name and namespace)
- /topology/tidb/basic-tidb-0.basic-tidb-peer.tidb-cluster.svc.cluster.local:4000/info (for across Kubernetes deployment, to match the TiDB sts name, namespace, and cluster domain)

Reply:
Refer to https://github.com/pingcap/tidb-operator/blob/master/pkg/manager/member/tikv_member_manager.go#L724.

Reply:
https://github.com/pingcap/tidb-operator/pull/3929/commits/42ea6a6bb8206d76a796ccdfbfb28a9734811747

Reply:
Suggest using `/topology/tidb/basic-tidb` as the prefix here so that we can get as least number of keys as we can.

Reply:
https://github.com/pingcap/tidb-operator/pull/3929/commits/0c142d5247143fac34492c7b1925413e42370282

Reply:
> Suggest using `/topology/tidb/basic-tidb` as the prefix here so that we can get as least number of keys as we can.

keep fetch all now since it will not come up with too much overhead and we can know all instance if need